src/main/java/frc/trigon/robot/subsystems/arm/simulationarm/SimulationArmIO.java

```diff
@@ -0,0 +1,104 @@
+package frc.trigon.robot.subsystems.arm.simulationarm;
+
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.subsystems.arm.ArmIO;
+import frc.trigon.robot.subsystems.arm.ArmInputsAutoLogged;
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.Logger;
+
+public class SimulationArmIO extends ArmIO {
+    private final SingleJointedArmSim angleMotor = SimulationArmIOConstants.ANGLE_MOTOR;
+    private final ElevatorSim elevatorMotor = SimulationArmIOConstants.ELEVATOR_MOTOR;
+    private final Mechanism2d armMechanism = SimulationArmIOConstants.ARM_MECHANISM;
+    private double
+            angleVoltage = 0,
+            elevatorVoltage = 0;
+
+    @Override
+    protected void updateInputs(ArmInputsAutoLogged inputs) {
+        angleMotor.update(RobotConstants.PERIODIC_TIME_SECONDS);
+        elevatorMotor.update(RobotConstants.PERIODIC_TIME_SECONDS);
+
+        inputs.angleMotorVoltage = angleVoltage;
+        inputs.angleMotorCurrent = angleMotor.getCurrentDrawAmps();
+        inputs.angleVelocityDegreesPerSecond = getAngleVelocityDegreesPerSecond();
+        inputs.anglePositionDegrees = Units.radiansToDegrees(angleMotor.getAngleRads());
+
+        inputs.elevatorMotorVoltage = elevatorVoltage;
+        inputs.elevatorMotorCurrent = elevatorMotor.getCurrentDrawAmps();
+        inputs.elevatorPositionRevolution = getElevatorPositionRevolutions();
+        inputs.elevatorVelocityRevolutionsPerSecond = elevatorMotor.getVelocityMetersPerSecond() / SimulationArmIOConstants.METERS_PER_REVOLUTION;
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        setAngleVoltage(0);
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        setElevatorVoltage(0);
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleVoltage(calculateAngleOutput(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorVoltage(calculateElevatorOutput(targetState));
+    }
+
+    @Override
+    protected void startAdvantageKitLogging(LogTable logTable) {
+        armMechanism.akitLog(logTable);
+        Logger.recordOutput("Arm", armMechanism);
+    }
```

Comment:
You missed my point. _Why_ are you doing this?
Read the [docs](https://docs.wpilib.org/en/stable/docs/software/dashboards/glass/mech2d-widget.html) again